source/user-management.rst

```diff
@@ -0,0 +1,220 @@
+
+Method 1: Manually Manage Users with MySQL Client
+-------------------------------------------------
+
+In order to manually manage users, you can issue ``CREATE USER`` and ``GRANT`` statements as you normally would with any MySQL client connected to the cluster running in the context of the Operator. 
+
+Our regular Docker image for Percona Server ``percona:5.7`` already contains a client which can be used for running BASH and the MySQL client or executing SQL statements directly with ``kubectl run``.
+
+An example of how you might use it to run a grant statement is as below:
+
+.. code:: bash
+
+    kubectl run -it --rm percona-client --image=percona:5.7 --restart=Never -- mysql -hcluster1-pxc -uroot -proot_password
+    mysql> GRANT ALL PRIVILEGES ON database1.* TO 'user1'@'%';
+
+You can refer to the official documentation to find out more about the `CREATE USER <https://dev.mysql.com/doc/refman/8.0/en/create-user.html>`_ and `GRANT <https://dev.mysql.com/doc/refman/8.0/en/grant.html>`_ statements.
+
+After you've created your users and issued grants, it's recommended to also manually sync these users to ProxySQL using ``proxysql-admin``.  The following one-liner can do this for you by running ``proxysql-admin --syncusers`` on the first ProxySQL instance in your cluster.
+
+.. code:: bash
+
+   kubectl exec -it cluster1-proxysql-0 -- proxysql-admin --config-file=/etc/proxysql-admin.cnf --syncusers
+
+You can validate a user synced successfully by connecting through ProxySQL with the MySQL client and then running a simple query.  A simple example of this is below:
+
+.. code:: bash
+
+   kubectl run -it --rm percona-client --image=percona:5.7 --restart=Never -- bash -il
+   percona-client:/$ mysql -h cluster1-proxysql -uuser1 -ppassword1
+   mysql> SELECT * FROM database1.table1 LIMIT 1;
+
+Method 2: Automatically Manage Users with Kubernetes Secret
+-----------------------------------------------------------
+
+YAML Object Format
+******************
+
+The YAML Object is stored as a `Kubernetes Secrets <https://kubernetes.io/docs/concepts/configuration/secret/>`_ object, containing two sets of data:
+
+1. Some **username:password** pairs that are compatible with `Kubernetes secretKeyRef reference <https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/#define-a-container-environment-variable-with-data-from-a-single-secret>`_,
+2. A **grants.yaml** field with a specially composed and formatted text.
+
+The text stored in ``grants.yaml`` field is a *YAML fragment*, which contains two *dictionaries of dictionaries*: 
+
+1. **roles dictionary** specifies role grants,
+2. **users dictionary** specifies user grants.
+
+The following example illustrates the whole YAML object:
+
+.. code:: yaml
+
+   apiVersion: v1
+   kind: Secret
+   metadata:
+     name: secrets-for-users
+     annotations:
+       status: applied
+       checksum:
```

Comment:
according to Kubernetes best practices custom labels and annotations should have domain name
```suggestion
       percona.com/users-status: applied
       percona.com/users-checksum:
```